test/lib/runtime/broker_test.exs

```diff
@@ -1238,4 +1281,72 @@ defmodule Ask.Runtime.BrokerTest do
     end)
   end
 
+  defp init_completed_dispositions_shouldnt_be_rejected_tests() do
+    insert_channel = fn mode ->
+      insert(:channel, settings: TestChannel.new() |> TestChannel.settings(), type: mode)
+    end
+
+    ivr_channel = insert_channel.("ivr")
+    sms_channel = insert_channel.("sms")
+
+    quiz = insert(:questionnaire, steps: @completed_dispositions_shouldnt_be_rejected_dummy_steps)
+
+    insert_survey = fn mode ->
+      insert(
+        :survey,
+        schedule: Schedule.always(),
+        state: "running",
+        questionnaires: [quiz],
+        mode: [[mode]],
+        count_partial_results: true,
+        quota_vars: ["City"]
+      )
+    end
+
+    ivr_survey = insert_survey.("ivr")
+    sms_survey = insert_survey.("sms")
+
+    insert_bucket = fn survey ->
+      insert(:quota_bucket, condition: %{City: "GBA"}, quota: 1, count: 0, survey: survey)
+    end
+
+    insert_bucket.(ivr_survey)
+    insert_bucket.(sms_survey)
+
+    insert_group = fn survey ->
+      insert(:respondent_group, survey: survey, respondents_count: 1) |> Repo.preload([:channels])
+    end
+
+    ivr_group = insert_group.(ivr_survey)
+    sms_group = insert_group.(sms_survey)
+
+    insert_group_channel = fn mode, channel, group ->
+      RespondentGroupChannel.changeset(%RespondentGroupChannel{}, %{
+        respondent_group_id: group.id,
+        channel_id: channel.id,
+        mode: mode
+      })
+      |> Repo.insert()
+    end
+
+    insert_group_channel.("ivr", ivr_channel, ivr_group)
+    insert_group_channel.("sms", sms_channel, sms_group)
+
+    %{
+      ivr_respondent: insert(:respondent, survey: ivr_survey, respondent_group: ivr_group),
+      sms_respondent: insert(:respondent, survey: sms_survey, respondent_group: sms_group)
+    }
+  end
+
+  defp broker_poll(), do: Broker.handle_info(:poll, nil)
+
+  defp respondent_reply(respondent_id, reply_message, mode) do
```

Comment:
Since this is a very helpful method, that is already being used in `retries_histogram_test`, perhaps we could move it to `Ask.TestHelpers` or into some of the support modules instead of having it duplicated 😬

Reply:
Totally! Changed! Thanks!!!